integration-tests/src/test/java/com/example/IntegrationTest.java

```diff
@@ -142,6 +142,27 @@
     assertThat(target.subtypeCalled).isTrue();
   }
 
+  /**
+   * [@Inject] Constructors are injected first, followed by fields, and then methods. Fields and methods in
+   * superclasses are injected before those in subclasses.
+   *
+   * @see javax.inject.Inject
+   */
+  @Test public void memberInjectionOrder() {
+    MemberInjectionOrder component = backend.create(MemberInjectionOrder.class);
+    MemberInjectionOrder.SubType target = new MemberInjectionOrder.SubType();
+    component.inject(target);
+    target.calls.add("injected: " + target);
```

Comment:
I don't think you need this or the booleans to track method calls. The logs already contain that information. All we need to see is constructor, base method, and sub method which will inform the field injection order based on the output.

Reply:
good point about booleans, came over with migration; re "injected", wanted to see `subCalled` being true, but that's just testing the fixture now that I look at it fresh